engines/mohawk/riven_vars.cpp

```diff
@@ -277,6 +277,22 @@ uint32 &MohawkEngine_Riven::getStackVar(uint32 index) {
 	return _vars[name];
 }
 
+namespace {
+
+uint32 getTwoRandomDomePositionsRng(Common::RandomSource *rnd, uint min, uint max) {
+	uint first = rnd->getRandomNumberRng(min, max);
+	uint second = rnd->getRandomNumberRng(min, max - 1);
+
+	// Avoid overlap of the two positions
+	if (second >= first) {
+		second++;
```

Comment:
I think this should be "second=max" instead if you want equal probability.

In that case, if `first==max`, then the condition will always fail, and if `first!=max`, then it will give equal probability to `second==max` as any other value.  As-is, this is giving slightly higher probability to `first+1`